src/rdf4cpp/rdf/datatypes/owl/Rational.cpp

```diff
@@ -0,0 +1,54 @@
+#include <rdf4cpp/rdf/datatypes/owl/Rational.hpp>
+
+namespace rdf4cpp::rdf::datatypes::registry {
+
+template<>
+capabilities::Default<owl_rational>::cpp_type capabilities::Default<owl_rational>::from_string(std::string_view s) {
+    if (auto pos = s.find_last_of('-'); pos != std::string_view::npos && pos != 0) {
+        // owl:rational only allows - at beginning
+        throw std::runtime_error{"owl:rational parsing error: invalid sign position"};
+    }
+
+    if (s.find_first_not_of("0123456789/-") != std::string_view::npos) {
+        throw std::runtime_error{"owl:rational parsing error: invalid character in string"};
+    }
```

Comment:
why are those checks done here? does the underlying type allow more than what is permitted by owl:rational?

Reply:
boost rational allows hex during parsing and i don't think there is any way to disable that

Reply:
added comments